chore/CheckJavadoc.java

```diff
@@ -0,0 +1,285 @@
+      }
+      if (isBlank(tag.getElements().subList(1, tag.getElements().size()))) {
+        violation(executable, ViolationType.MISSING_PARAMETER_DESCRIPTION, parameterName);
+      }
+    }
+
+    for (String param : params) {
+      violation(executable, ViolationType.MISSING_PARAMETER, param);
+    }
+  }
+
+  private void violationMissingReturn(CtExecutable<?> executable, JavadocCommentView doc) {
+    if (executable.getType().equals(executable.getFactory().Type().voidPrimitiveType())) {
+      return;
+    }
+    List<JavadocElement> returnTagValues = new ArrayList<>(doc.getBlockTag(StandardJavadocTagType.RETURN));
+    for (JavadocElement element : doc.getBody()) {
+      if (element instanceof JavadocInlineTag inline && inline.getTagType() == StandardJavadocTagType.RETURN) {
+        returnTagValues.add(element);
+      }
+    }
+
+    if (returnTagValues.isEmpty()) {
+      violation(executable, ViolationType.MISSING_RETURN);
+      return;
+    }
+    // TODO: Disallow more than one return tag
+    JavadocElement tag = returnTagValues.get(0);
+    if (isBlank(tag)) {
+      violation(executable, ViolationType.MISSING_RETURN_DESCRIPTION);
+    }
+  }
+
+  private void violation(CtExecutable<?> context, ViolationType type, Object... args) {
+    String message = type.getTemplate().formatted(args);
+    Path fileAsPath = context.getPosition().getFile().toPath();
+    String file = fileAsPath.startsWith(CWD) ? CWD.relativize(fileAsPath).toString() : fileAsPath.toString();
+    int line = context.getPosition().getLine();
+    int column = context.getPosition().getColumn();
+    String methodName = context.getSimpleName() + context.getParameters()
+      .stream()
+      .map(it -> it.getType().getSimpleName())
+      .collect(Collectors.joining(",", "(", ")"));
+
+    violationCounts.merge(type, 1, Math::addExact);
+
+    System.out.println("[ERROR] " + file + ":" + line + ":" + column + ": " + message + " in " + methodName);
+  }
+
+  private static boolean isBlank(JavadocElement element) {
+    return element.accept(new IsBlankVisitor());
+  }
+
+  private static boolean isBlank(Collection<JavadocElement> elements) {
+    return elements.stream().allMatch(CheckJavadoc::isBlank);
+  }
+
+  public static void main(String... args) {
+    Launcher launcher = new Launcher();
+    launcher.getEnvironment().setComplianceLevel(17);
+    launcher.addInputResource("src/main/java");
+    launcher.addInputResource("spoon-javadoc/src/main/java");
+    CtModel model = launcher.buildModel();
+
+    CheckJavadoc checker = new CheckJavadoc();
+    checker.check(model, launcher.getFactory());
+
+    System.out.println("Violations per category");
+    for (var entry : checker.violationCounts.entrySet().stream().sorted(Entry.comparingByValue()).toList()) {
+      System.out.println(entry);
+    }
+
+    int errorCount = checker.violationCounts.values().stream().mapToInt(Integer::intValue).sum();
```

Comment:
I hope we don't encounter an overflow here \:)

Reply:
I hope we just delete the check if we have an overflow.

Reply:
I would delete spoon tbh. At that point we had a good run, but you sometimes have to admit defeat.